### PR TITLE
Feat(transfer investigation) 

### PR DESCRIPTION
**Pull request by nevobonneh**
This PR implements the following feature [1175](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1075/)
In this PR we added Tabs to TransferInvestigation option on TableFooter. One Tab for  multi transfering between Desks (this was the option that already exists) and the other Tab for multi Transfering between Counties.

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    allDesks={allDesks}
+                />
+            </TabPanel>
+        </div>
+    );
+};
+
+interface TabPanelProps {
+    children?: React.ReactNode;
+    index: any;
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx:84]: why any?

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    value: any;
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx:85]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                {value === index && (
+                    <Box p={3}>
+                        {children}
+                    </Box>
+                )}
+            </div>
+        );
+    }
+
+    const { allDesks, allCounties, onClose, onDeskTransfer, onCountyTransfer } = props;
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx:42]: maybe move it to the top of the component

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 4]
```diff
@@ -583,6 +581,23 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
         }
     };
 
+    const changeGroupsCounty = async (groupIds: string[], newSelectedCounty: County | null, transferReason: string) => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:584]: you used async but didn't use await

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 5]
```diff
@@ -603,27 +618,44 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
         }
     }
 
-    const getDefaultCellStyles = (cellKey : string) => {
+    const changeInvestigationCounty = async (epidemiologyNumbers: number[], newSelectedCounty: County | null, transferReason: string) => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:621]: same

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 6]
```diff
@@ -219,32 +220,32 @@ usersRoute.get('/group', adminMiddleWare, (request: Request, response: Response)
...
-
-    graphqlRequest(UPDATE_COUNTY_BY_USER, response.locals, {
-        epidemiologyNumber: request.body.epidemiologyNumber,
-        newUser: userAdmin
-    }).then((result: any) => {
-        if (result?.data && !result?.errors) {
-            changeCountyLogger.info('The investigation county changed successfully', Severity.LOW);
-            response.send({message: 'The county has changed successfully'});
+    changeCountyLogger.info('launching graphql API counties request', Severity.LOW);
+    Promise.all(epidemiologyNumbers.map(epidemiologyNumber => graphqlRequest(UPDATE_COUNTY_BY_USER, response.locals, {
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:232]: why mapping the epidemiologyNumbers and execute UPDATE_COUNTY_BY_USER for each epidemiologyNumber
why not changing the query (trying to use IN syntax in sql)

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+import County from 'models/County';
+
+import TransferInvestigationDesk from './TransferInvestigationDesk';
+import TransferInvestigationCounty from './TransferInvestigationCounty';
+
+const TransferInvestigationTabs = (props: Props) => {
+
+    const { allDesks, allCounties, onClose, onDeskTransfer, onCountyTransfer } = props;
+
+    const StyledTab = withStyles((theme) =>
```
[the remark this answers is not in the document]
Comment (NaamanTsur) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationTabs.tsx:14]: Not necessarily a separate file.

**Review by NaamanTsur — APPROVED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Pull request merged**